dask/dataframe/core.py

```diff
@@ -322,7 +322,7 @@ def __init__(self, dsk, name, meta, divisions):
                 f"{typename(type(meta))}"
             )
         self._meta = meta
-        self._divisions = tuple(divisions)
+        self.divisions = tuple(divisions)
```

Comment:
This was committed accidentally while debugging why cudf was hitting a `RecursionError`, but I think I'll leave it (provided tests pass). The only reason I see to remove it is for performance issues, but dask dataframe already has enough overhead that this should be negligible.

Reply:
+1 on more validation wherever appropriate